bin.js

```diff
@@ -427,11 +427,26 @@ function catchify (asyncFn) {
   }
 }
 
+function deltams (now) {
+  const delta = process.hrtime(now)
+  return delta[0] * 1e3 + delta[1] * 1e-6
+}
+
 function checkArgs (args, help, version) {
   if (args[0] === '--' && args[1] !== 'node') {
     printHelp(help, version)
     process.exit(1)
   }
+  // Check for timeout usage and set the timeout delay
+  const delayused = args.toString().includes('timeout')
+
+  if (delayused) {
+    const timeout = args.filter(a => a.includes('timeout'))
+    const delay = timeout.toString().split('=')[1]
+
+    const now = process.hrtime()
+    while (deltams(now) < delay) { }
+  }
```

Comment:
Blocking synchronously with while loop will cpu block and that's not what we want here. The better approach would be to use `setTimeout()` to defer execution for the given amount of time; or, if we absolutely need a sync block, use `Atomics.wait()` to block without spinning the CPU.

Reply:
@jasnell - I had a chat with @goto-bus-stop around the structure of Node Clinic. I'll close this PR/branch and take a new approach.